File: 0290-word-pattern/0290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        d = {}
        ls = s.split()
        
        if len(pattern) != len(ls):
            return False
        
        for i in range(len(pattern)):
            if pattern[i] not in d.keys():
                if ls[i] not in d.values():
                    d[pattern[i]] = ls[i]
                else:
                    return False
            elif d[pattern[i]] != ls[i]:
                    return False
                
        return True

File: 0290-word-pattern/0290-word-pattern.java
class Solution {
    public boolean wordPattern(String pattern, String s) {
        String[] ls = s.split(" ", -1);
        
        if (pattern.length() != ls.length) return false;
        
        HashMap<Character, String> map = new HashMap<>();
        
        for (int i=0; i<pattern.length(); i++){
            char curr = pattern.charAt(i);
            if (!map.containsKey(curr)){
                if (!map.containsValue(ls[i])){
                    map.put(curr, ls[i]);
                }
                else{
                    return false;
                }
            }
            else if (!map.get(curr).equals(ls[i])) return false;
        }
        return true;
    }
}